pkg/virt-launcher/virtwrap/api/converter_test.go

```diff
@@ -551,36 +553,131 @@ var _ = Describe("Converter", func() {
 			Expect(vmiToDomainXMLToDomainSpec(vmi, c).Type).To(Equal(domainType))
 		})
 
-		It("should convert CPU cores and model", func() {
-			v1.SetObjectDefaults_VirtualMachineInstance(vmi)
-			vmi.Spec.Domain.CPU = &v1.CPU{
-				Cores: 3,
-				Model: "Conroe",
-			}
-			domainSpec := vmiToDomainXMLToDomainSpec(vmi, c)
+		Context("when CPU spec defined", func() {
+			It("should convert CPU cores and model", func() {
+				v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+				vmi.Spec.Domain.CPU = &v1.CPU{
+					Cores:   3,
+					Sockets: 2,
+					Threads: 2,
+					Model:   "Conroe",
+				}
+				domainSpec := vmiToDomainXMLToDomainSpec(vmi, c)
 
-			Expect(domainSpec.CPU.Topology.Cores).To(Equal(uint32(3)))
-			Expect(domainSpec.CPU.Topology.Sockets).To(Equal(uint32(1)))
-			Expect(domainSpec.CPU.Topology.Threads).To(Equal(uint32(1)))
-			Expect(domainSpec.CPU.Mode).To(Equal("custom"))
-			Expect(domainSpec.CPU.Model).To(Equal("Conroe"))
-			Expect(domainSpec.VCPU.Placement).To(Equal("static"))
-			Expect(domainSpec.VCPU.CPUs).To(Equal(uint32(3)))
-		})
+				Expect(domainSpec.CPU.Topology.Cores).To(Equal(uint32(3)))
+				Expect(domainSpec.CPU.Topology.Sockets).To(Equal(uint32(2)))
```

Comment:
Please do add the labels here. Otherwise failure, if any will report:
Expected 3 to be 2. Which is not helpful. I would like to see
Expected sockets:
3 to be 2.

Written as `Expect(domainSpec.CPU.Topology.Sockets).To(Equal(uint32(2)), "Expect sockets")`

Please to this for the rest.